Wiki.md

```diff
@@ -0,0 +1,152 @@
+
+Here are a few points about the [`revert-web-wiki-updates.yml`](https://github.com/oppia/oppia/blob/develop/.github/workflows/revert-web-wiki-updates.yml) workflow:
+
+* The workflow uses the `oppia/get-github-app-token` action to retrieve an authentication token using the synchronizer bot's private key. We specify the version of the action using a git hash as an extra security precaution since the `oppia/get-github-app-token` repository won't be as closely watched as `oppia/oppi`. This means that even if `oppia/get-github-app-token` is compromised, any malicious code added there won't be pulled into `oppia/oppia`.
+* We configure the git name as `oppia-wiki-synchronizer[bot]` and email as `102317631+oppia-wiki-synchronizer[bot]@users.noreply.github.com`. This choice of name and email causes GitHub to show the app as the commit author, though this feature is not documented by GitHub.
+* When we push to the deployment repository, we run `git push source master:develop`. The `master:develop` says to push the local `master` branch to the remote `develop` branch. We have different branch names because at Oppia we generally use `develop` as our main development branch, but GitHub requires that the wiki deployment repository main branch be called `master` (this detail is also not documented).
+
+### Security analysis
+
+This approach introduces the following security concerns:
+
+* The permissions for the documentation source repository must be maintained as they no longer automatically follow the deployment repository permissions.
+* Access to the GitHub App must be secured. In particular, the app should be created at the organization level, and the minimum number of people should be granted [app manager permissions](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization#github-app-managers). Organization owners automatically have access.
+* The GitHub App's credentials must also be secured. In particular, there should be no copies of the private key outside of GitHub actions secrets.
+
+Here are some alternative approaches:
+
+* Creating a new user account that we can give access to the necessary repositories and who the scripts will act as.
+
+  * This means we have to keep track of securing another account and that account's credentials. The GitHub App approach is better because we can use existing GitHub accounts to manage the app, and it's not possible to log in as the app.
+  * The app's actions are clearly marked as being by a bot, which is better than a user account which will look like a human user.
+
+* Using a personal access token (PAT).
+
+  * The PAT grants broad access across all the repositories the user who generated the PAT has privileges for. This introduces a risk that a compromise of Oppia could grant an attacker access to other, unaffiliated repositories. A GitHub App has more fine-grained permissions that can be restricted to particular repositories.
```

Comment:
```suggestion
  * The PAT grants broad access across all the repositories for the user who generated the PAT has privileges for. This introduces a risk that a compromise of Oppia could grant an attacker access to other, unaffiliated repositories. A GitHub App has more fine-grained permissions that can be restricted to particular repositories.
```

Reply:
This text doesn't scan with the suggested change

Reply:
Does this work better?

The PAT generated by a user gives broad access across all the repositories where the user has privileges.

Reply:
done